### PR TITLE
issue/6096-dot-org-always-pending-7.7

### LINKED ISSUE #6096: .org: New Posts are always posted Pending Review
### Expected behavior
When I create a new post and submit it, it should be submitted as "Published". Also, when I modify a post status to "Published" this should be reflected.

### Actual behavior
New posts and modified post statuses are published as "Pending Review". The only way to get a post into the "Published" state is to use the "Publish" function on the posts listing screen after tapping on "More".

### Steps to reproduce the behavior
Create a new post, add a headline and use the "Submit" button in the header (note that on WordPress.com and Jetpack blogs this is a "Publish" button).

##### Tested on Xperia X Compact, Android 7.0, WordPress alpha-60

This is on a WordPress 4.8 install without Jetpack.

### PR DESCRIPTION
**Pull request by nbradbury**
Fixes #6096 

To test: publish a post from a .org site. Prior to this fix it would show as PENDING, it should now show as PUBLISHED.

### CODE REVIEW

**Comment by nbradbury** [comment]
@daniloercoli or @aforcier - one of you want to review this?

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java [hunk 1]
```diff
@@ -942,6 +942,18 @@ public boolean isNewPost() {
         return mIsNewPost;
     }
 
+    /*
+     * returns true if the user has permission to publish the post - assumed to be true for
+     * dot.org sites because we can't retrieve their capabilities
+     */
+    private boolean canPublishPost() {
```
Comment (aforcier) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: May I suggest calling this `userCanPublishPosts`? There are lots of nuances to whether a post can be published (e.g. we have [`isPublishable` checks](https://github.com/wordpress-mobile/WordPress-Android/blob/15c3cba2e51a79a931d543401c91a8422246f509/WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java#L1083) in several places), and I think we should keep things as specific as possible.

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java [hunk 2]
```diff
@@ -942,6 +942,18 @@ public boolean isNewPost() {
         return mIsNewPost;
     }
 
+    /*
+     * returns true if the user has permission to publish the post - assumed to be true for
+     * dot.org sites because we can't retrieve their capabilities
```
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: Just tested with a Contributor role on a .org site. The UX is fine (user gets a error after publishing, and error message looks good)
screen shot 2017-06-20 at 18 41 12

**Comment by aforcier** [comment]
Tested:
* .org site, user can publish posts ✅
* .org site, user can't publish posts ✅ ❓ (post uploads, and fails, with a descriptive error as [shown by @maxme](https://github.com/wordpress-mobile/WordPress-Android/pull/6113#discussion_r123031066)
* WP.com site, user can publish posts ✅
* WP.com site, user can't publish posts ✅ (get 'Submit' and post is saved as pending)
* Jetpack site over WP.com, user can publish posts ✅ (same as WP.com really)

Just one [small naming suggestion](https://github.com/wordpress-mobile/WordPress-Android/pull/6113#pullrequestreview-45194623) and we should be good to go @nbradbury.

**Comment by aforcier** [comment]
:shipit:

**Comment by gingerling** [comment]
Hi, this code was merged but I think a bug it caused was not resolved - the bug has been open for 17 months now and has been causing a lot of problems for the organisations I help - these are charities who want to give access for volunteers to work on the site by making posts but need to check the content before it's published.

Here is the bug: https://github.com/wordpress-mobile/WordPress-Android/issues/6011

Can anyone help?

**Pull request merged**